velox/exec/PartitionedOutputBufferManager.cpp

```diff
@@ -544,7 +544,14 @@ void PartitionedOutputBufferManager::getData(
     uint64_t maxBytes,
     int64_t sequence,
     DataAvailableCallback notify) {
-  getBuffer(taskId)->getData(destination, maxBytes, sequence, notify);
+  std::shared_ptr<PartitionedOutputBuffer> buffer;
```

Comment:
@karteekmurthys 

There are multiple places that call getBuffer. It is not clear why this particular place should handle the exception and others should not. Would you clarify?

Similar to the other PR, we should avoid programming-by-exception pattern and instead introduce a method getBufferIfExists that return nullptr if buffer doesn't exist. This method can then be used whenever the caller wants to handle missing buffer (i.e. deleteResults and maybe here).

The method documentation needs to be updated to describe the new behavior and a test needs to be added.